test/helper.rb

```diff
@@ -8,10 +8,35 @@
 require 'mocha/minitest'
 require 'phenix'
 
-RAILS_ENV = 'test'
+ENV['RAILS_ENV'] = 'test'
+ENV['LEGACY_CONNECTION_HANDLING'] = 'true' if ENV['LEGACY_CONNECTION_HANDLING'].nil?
+
+if ActiveRecord.version >= Gem::Version.new('6.1')
+  ActiveRecord::Base.legacy_connection_handling = (ENV['LEGACY_CONNECTION_HANDLING'] == 'true')
+end
```

Comment:
Should we add a “helper” constant here?
```suggestion
end

RAILS_6_1_WITH_NEW_CONNECTION_HANDLING = ActiveRecord.version >= Gem::Version.new('6.1') && !ActiveRecord::Base.legacy_connection_handling
```

That would make it easier to write the conditional code elsewhere.